### PR TITLE
Warning for mismatching env variables

### PR DESCRIPTION
**Pull request by yepninja**
Warning for a minor release.

### CODE REVIEW

File: src/configuration_env.js [hunk 1]
```diff
@@ -13,12 +13,32 @@
...
     return (obj && obj[key]) || defaultValue;
   }
 
   static _getSamplerFromEnv(config) {
+
+    Object
+      .keys(wrongEnvVars)
+      .forEach(env => {
+        if (process.env[env]) {
+          console.warn(`You are using wrong missmatching env variable ${env}. Use ${wrongEnvVars[env]} instead. \nMismatching env variable will be removed in the next major release (4.x.x)`)
```
Comment (yepninja) [inline src/configuration_env.js]: Maybe better to collect warnings, and than log them with the logger. But not everybody uses logger.

Comment (yurishkuro) [reply src/configuration_env.js]: Console is fine, I think. Replace “wrong mismatching” with “deprecated”

File: src/configuration_env.js [hunk 2]
```diff
@@ -13,12 +13,32 @@
 import Configuration from './configuration.js';
 import Utils from './util.js';
 
+const wrongEnvVars = {
```
Comment (yurishkuro) [inline src/configuration_env.js]: DeprecatedEnvVars

File: src/configuration_env.js [hunk 3]
```diff
@@ -13,12 +13,32 @@
...
+  JAEGER_SAMPLER_HOST: 'JAEGER_SAMPLER_MANAGER_HOST_PORT',
+  JAEGER_SAMPLER_PORT: 'JAEGER_SAMPLER_MANAGER_HOST_PORT',
+  JAEGER_REPORTER_ENDPOINT: 'JAEGER_ENDPOINT',
+  JAEGER_REPORTER_USER: 'JAEGER_USER',
+  JAEGER_REPORTER_PASSWORD: 'JAEGER_PASSWORD',
+  JAEGER_REPORTER_AGENT_HOST: 'JAEGER_AGENT_HOST',
+  JAEGER_REPORTER_AGENT_PORT: 'JAEGER_AGENT_PORT',
+  JAEGER_DISABLE: 'JAEGER_DISABLED'
+};
```
Comment (yurishkuro) [inline src/configuration_env.js]: Why does this need to be a map and not an array?

Comment (yurishkuro) [reply src/configuration_env.js]: Or, nvm

File: src/configuration_env.js (hunk 2 above)
Comment (yepninja) [reply src/configuration_env.js]: Done

File: src/configuration_env.js (hunk 1 above)
Comment (yepninja) [reply src/configuration_env.js]: Done

File: src/configuration_env.js [hunk 4]
```diff
@@ -13,12 +13,32 @@
...
     return (obj && obj[key]) || defaultValue;
   }
 
   static _getSamplerFromEnv(config) {
+
+    Object
+      .keys(deprecatedEnvVars)
+      .forEach(env => {
+        if (process.env[env]) {
+          console.warn(`You are using deprecated env variable ${env}. Use ${deprecatedEnvVars[env]} instead. \nMismatching env variable will be removed in the next major release (4.x.x)`)
```
Comment (yurishkuro) [inline src/configuration_env.js]: s/Mismatching/Deprecated in the second sentence as well.

File: src/configuration_env.js [hunk 5]
```diff
@@ -13,12 +13,32 @@
...
+};
+
 export default class ConfigurationEnv {
   static _getConfigValue(obj, key, defaultValue) {
     return (obj && obj[key]) || defaultValue;
   }
 
   static _getSamplerFromEnv(config) {
+
+    Object
```
Comment (yurishkuro) [inline src/configuration_env.js]: why is this check inside `_getSamplerFromEnv` function?

Comment (yepninja) [reply src/configuration_env.js]: Node.js gets env variables in runtime (call of `getenv` for each variable). So check is when process.env is used.

Related issues:
- https://github.com/nodejs/node/issues/3104
- https://github.com/nodejs/node/issues/22960

File: src/configuration_env.js (hunk 4 above)
Comment (yepninja) [reply src/configuration_env.js]: Fixed

**Comment by yepninja** [comment]
Opps. Almost there is no habit to code without eslint.
The build is fixed!

**Comment by yepninja** [comment]
Nope. That issue is related to PR #330

**Pull request merged**